src/azure-cli/azure/cli/command_modules/serviceconnector/action.py

```diff
@@ -118,7 +118,11 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
             raise ValidationError('Usage error: {} [KEY=VALUE ...]'.format(option_string))
         d = {}
         for k in properties:
-            raise ValidationError('Unsupported Key {} is provided for parameter --system-identity')
+            v = properties[k]
+            if k.lower() == 'mysql-identity-id':
+                d['mysql-identity-id'] = v[0]
```

Comment:
we may also need deal with the help message of `--system-identity`. Otherwise users may never know he can use `--system-identity mysql-identity-id=xx`.

And it's better we only extend this for `mysql/mysql-flexible`, and let other targets keep their original behavior